zio-http/shared/src/main/scala/zio/http/endpoint/Description.scala

```diff
@@ -0,0 +1,11 @@
+package zio.http.endpoint
+
+import zio.http.codec.Doc
+
+final case class Description(doc: Doc, tags: List[String]) {
```

Comment:
I know this form is more convenient to work with as an implementor, but I feel like just adding another type of `Doc`, e.g. `Doc.Tagged(doc, tag)` will result in fewer breaking changes and a more ergonomic API for the developer, who ordinarily doesn't want tags. Then you can define `Doc#tags` to extract `List[String]` through recursion and otherwise use the same as you use this one.

Reply:
I get your point, but I am not sure how to handle tags in the middle of a docs ADT tree. Would we just ignore it, when rendering html/md? Do I get also deep nested tags when using recursion on any doc?

These thought lead me to this design

Reply:
Btw: Your design might be better, if not only endpoints but maybe data structures need to be tagged.

Reply:
I would collect them and render as `<ul> / <li>`.

So the tags are associated with the endpoint, and not specific items inside the endpoint?

Reply:
> So the tags are associated with the endpoint, and not specific items inside the endpoint?
I am not sure. For OpenAPI that is the only usecase (afaik). But I think Smithy has tags also on items.

I'll do changes later, I'll be working today on some of the larger issues